features/nanostack/nanostack-interface/Nanostack.cpp

```diff
@@ -816,6 +816,10 @@ nsapi_error_t Nanostack::getsockopt(void *handle, int level, int optname, void *
 
     NanostackLockGuard lock;
 
+    if (level == NSAPI_SOCKET) {
```

Comment:
I think this could be done by better translation of the return code from Nanostack below, same for `socket_setsockopt` above). As I recall, it properly distinguishes two codes (-2 and -3?) for "bad parameter" and "unsupported option".

If you had that translation correct below, then you wouldn't need this extra clause here.